src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php

```diff
@@ -172,7 +174,16 @@ public function update_value( $meta, $object_id ) {
 				continue;
 			}
 
-			$value = $meta[ $name ];
+			/*
+			 * Non-single meta can also be removed by sending an empty array.
+			 */
+			if ( is_array( $value ) && empty( $value ) && ! $args['single'] ) {
```

Comment:
Could we put this in the same branch as the existing `is_null` check? We could also simplify the test to `array() === $value`

Reply:
If that expression would still be too awkward, we could move the check to a variable before hand. But I'd like to try and keep it to one exit point to simplify https://core.trac.wordpress.org/ticket/48823.